Compilation error when using JsonUnwrapped and wildcard response type

### Expected Behavior

No compilation error.

### Actual Behaviour

Compilation error with the following stacktrace:
```
Caused by: java.lang.NullPointerException: Cannot invoke "io.swagger.v3.oas.models.media.Schema.getProperties()" because "wrappedPropertySchema" is null
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.handleUnwrapped(AbstractOpenApiVisitor.java:1051)
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.processSchemaProperty(AbstractOpenApiVisitor.java:1091)
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.processPropertyElements(AbstractOpenApiVisitor.java:2570)
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.populateSchemaProperties(AbstractOpenApiVisitor.java:2463)
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.getSchemaDefinition(AbstractOpenApiVisitor.java:1966)
	at io.micronaut.openapi.visitor.AbstractOpenApiVisitor.resolveSchema(AbstractOpenApiVisitor.java:983)
	at io.micronaut.openapi.visitor.AbstractOpenApiEndpointVisitor.buildContent(AbstractOpenApiEndpointVisitor.java:1980)
	at io.micronaut.openapi.visitor.AbstractOpenApiEndpointVisitor.addResponseContent(AbstractOpenApiEndpointVisitor.java:1195)
	at io.micronaut.openapi.visitor.AbstractOpenApiEndpointVisitor.readResponse(AbstractOpenApiEndpointVisitor.java:1180)
	at io.micronaut.openapi.visitor.AbstractOpenApiEndpointVisitor.visitMethod(AbstractOpenApiEndpointVisitor.java:434)
	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.visitMethod(TypeElementVisitorProcessor.java:331)
	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.visitClass(TypeElementVisitorProcessor.java:316)
	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.process(TypeElementVisitorProcessor.java:262)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1023)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:939)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1267)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1382)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1234)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:916)
```

### Steps To Reproduce

Use the following test:

```groovy
void "test issue with JsonUnwrapped and wildcard response type"() {

        given:"An API definition"
        when:
        buildBeanDefinition('test.MyBean', '''
package test;

import io.reactivex.*;
import io.micronaut.http.annotation.*;
import com.fasterxml.jackson.annotation.*;

@Controller("/test")
interface TestOperations {

    @Post("/")
    Single<Test<?>> save(String name, int age);
}

class Base {
}

class Test<T extends Base> {
    @JsonUnwrapped
    public T wrapped;
}

@jakarta.inject.Singleton
class MyBean {}
''')
        then:"the state is correct"
        Utils.testReference != null

        when:"The OpenAPI is retrieved"
        OpenAPI openAPI = Utils.testReference
        Schema schema = openAPI.components.schemas['Test_Base_']

        then:"the components are valid"
        schema.type == 'object'
    }
```

### Environment Information

_No response_

### Example Application

_No response_

### Version

micronaut-openapi 6.11.0

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
@@ -214,6 +214,7 @@ import static io.micronaut.openapi.visitor.StringUtil.OPEN_BRACE;
 import static io.micronaut.openapi.visitor.StringUtil.SLASH;
 import static io.micronaut.openapi.visitor.StringUtil.UNDERSCORE;
 import static io.micronaut.openapi.visitor.Utils.isOpenapi31;
+import static io.micronaut.openapi.visitor.Utils.resolveOpenApi;
 import static java.util.stream.Collectors.toMap;
 
 /**
@@ -1172,9 +1173,14 @@ abstract class AbstractOpenApiVisitor {
     private void handleUnwrapped(VisitorContext context, Element element, ClassElement elementType, Schema<?> parentSchema, AnnotationValue<JsonUnwrapped> uw) {
         Map<String, Schema> schemas = SchemaUtils.resolveSchemas(Utils.resolveOpenApi(context));
         ClassElement customElementType = getCustomSchema(elementType.getName(), elementType.getTypeArguments(), context);
+        var elType = customElementType != null ? customElementType : elementType;
         String schemaName = stringValue(element, io.swagger.v3.oas.annotations.media.Schema.class, PROP_NAME)
-            .orElse(computeDefaultSchemaName(null, customElementType != null ? customElementType : elementType, elementType.getTypeArguments(), context, null));
+            .orElse(computeDefaultSchemaName(null, elType, elementType.getTypeArguments(), context, null));
         Schema<?> wrappedPropertySchema = schemas.get(schemaName);
+        if (wrappedPropertySchema == null) {
+            getSchemaDefinition(resolveOpenApi(context), context, elType, elType.getTypeArguments(), element, Collections.emptyList(), null);
+            wrappedPropertySchema = schemas.get(schemaName);
+        }
         Map<String, Schema> properties = wrappedPropertySchema.getProperties();
         if (CollectionUtils.isEmpty(properties)) {
             return;

--- a/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiJsonUnwrappedsSpec.groovy
+++ b/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiJsonUnwrappedsSpec.groovy
@@ -8,7 +8,7 @@ class OpenApiJsonUnwrappedsSpec extends AbstractOpenApiTypeElementSpec {
 
     void "test JsonUnwrapped annotation"() {
 
-        given:"An API definition"
+        given: "An API definition"
         when:
         buildBeanDefinition('test.MyBean', '''
 package test;
@@ -91,16 +91,16 @@ class Test {
 @jakarta.inject.Singleton
 class MyBean {}
 ''')
-        then:"the state is correct"
+        then: "the state is correct"
         Utils.testReference != null
 
-        when:"The OpenAPI is retrieved"
+        when: "The OpenAPI is retrieved"
         OpenAPI openAPI = Utils.testReference
         Schema schema = openAPI.components.schemas['Test']
         Schema dummySchema = openAPI.components.schemas['Dummy']
         Schema petSchema = openAPI.components.schemas['Pet']
 
-        then:"the components are valid"
+        then: "the components are valid"
         schema.type == 'object'
         schema.properties.size() == 13
         schema.properties['plain'].$ref == '#/components/schemas/Dummy'
@@ -175,5 +175,51 @@ class MyBean {}
         !exampleSchema.properties.nameInPojo
         exampleSchema.properties.nameInJson.type == 'string'
         exampleSchema.properties.nameInJson.description == 'example field'
+    }
+
+    void "test issue with JsonUnwrapped and wildcard response type"() {
+
+        given: "An API definition"
+        when:
+        buildBeanDefinition('test.MyBean', '''
+package test;
+
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Post;
+import io.reactivex.Single;
+
+@Controller("/test")
+interface TestOperations {
+
+    @Post
+    Single<Test<?>> save(String name, int age);
+}
+
+class Base {
+    
+    public String name;
+}
+
+class Test<T extends Base> {
+    @JsonUnwrapped
+    public T wrapped;
+}
+
+@jakarta.inject.Singleton
+class MyBean {}
+''')
+        then: "the state is correct"
+        Utils.testReference != null
+
+        when: "The OpenAPI is retrieved"
+        def openApi = Utils.testReference
+        Schema schema = openApi.components.schemas.Test_Base_
+
+        then: "the components are valid"
+        schema.type == 'object'
+        schema.properties
+        schema.properties.size() == 1
+        schema.properties.name
     }
 }

